test_cases/README.md

```diff
@@ -52,6 +52,15 @@ annotations,
 [this page](https://typing.readthedocs.io/en/latest/source/quality.html#testing-using-assert-type-and-warn-unused-ignores)
 provides a useful guide.
 
+### Naming convention
+
+Use the same top-level name for the module / package you would like to test.
+Use `test_${thing}.py` naming pattern for individual test files.
+For example:
+- Tests for `gather` defined in `stdlib/asyncio/tasks.pyi` will go into `stdlib/asyncio/test_gather.py`
+- Tests for `ExitStack` defined in `stdlib/contextlib.pyi` will go into `stdlib/test_contextlib.py`
```

Comment:
These examples don't show why ExitStack goes into `test_contextlib` directly, but `pow` doesn't go into `test_builtins`. We should add something like "By default, tests go into a test file with the same name as the stub file, prefixed with `test_`. For example: test_contextlib. If that file becomes too big, we instead create a directory with files named after individual objects being tested. For example: builtins/test_dict."